### DISCUSSION TITLE
Generating .po files

### DISCUSSION THREAD

**Discussion by tcastanheira** [wagtail/wagtail-localize#319] [Q&A] [answered]
When exporting .po files, shouldn't it be possible to export one single .po file for a given locale?

Or even as an alternative, shouldn't we be able to take advantage of utils like msgcat to concatenate multiple .po's?
(In this case, the UUID could become part of the comments before each translation instead of a file top level header?)

Thanks for all the help

**Comment by kaedroho** [comment, marked as answer]
Thanks @heav1811!

I was thinking of implementing something like this by putting all the PO files into a single zip file. So you would only have to download one file per locale and the pages are still separated. Would something like that suffice?

**Reply by tcastanheira** [reply]
Although this is a solution, it is not the best one, at least from my POV.

Even if you only have to submit the same folder for upload, same translations strings will need to be translated multiple times, creating a bit of redundant work.

From my perspective, a single .po file with the UUID's to which the translation is part of, in a comment before the msgtxt could easily be used to map the translatable content. 

My suggestion would be:
The scope of each locale in one .po file, multiple translations are only translated once, the import system can then abstract the rest.
Example:

```
#TranslationID: 29edbaea-2c6f-4dd1-a08a-b961b377e570
msgctxt "title"
msgid "..."
msgstr ""


#TranslationID: 29edbaea-2c6f-4dd1-a08a-asdasdasdasda
msgctxt "slug"
msgid "..."
msgstr ""

#TranslationID: 29edbaea-2c6f-4dd1-a08a-asdasdasdasda
#TranslationID: 29edbaea-2c6f-4dd1-a08a-b961b377e570
msgctxt "slug2"
msgid "Multiple page translation"
msgstr ""

```
Maybe the odds of something like this happening are small.
What are your thoughts on this?

**Reply by kaedroho** [reply]
> Even if you only have to submit the same folder for upload, same translations strings will need to be translated multiple times, creating a bit of redundant work.

There might be cases where someone would want different translations for the two pages, even if the source text is the same. I think the "translation memory" feature that exists in a lot of translation tools is a good solution for the redundant work problem.
